### DISCUSSION TITLE
Grooming/petting minigame and friendship

### DISCUSSION THREAD

**Discussion by HippasusTwo** [Tuxemon/Tuxemon#2263] [Ideas]
I think it might be nice to have at some point some kind of friendship-increasing minigame, and to make it relevant there could be friendship specific evolutions of some creatures.
The easiest way to go about it would just be to have the default sprite (so one wouldn't have to make more), a brush sprite, and a little ball / cat-toy sprite, and little emoticons that pop over the critter's sprite.

Depending on how fusion is implemented, this could actually go along well with such a minigame, because the code implementation I'm familiar with uses separated out body parts like a head and information about where the head is, so that fusion can be done by simply swapping heads on to different bodies. Obviously, knowing where on a sprite the head is supposed to be would be good for a grooming minigame.

I also think it would spice things up if there were alternative capture methods/scenarios, like randomly encountering every once in awhile a tuxemon who gives off the message 'It doesn't seem to be very interested in fighting at the moment. Maybe you can impress it in some other way?', and can offer food and toys depending on a randomized personality to it - all normal capture devices used on it will fail until this is done, instead of reacting to lowered health.

(Out of curiosity, how easy is it to notice a new thread in the forum has been posted?)

**Comment by Sanglorian** [comment]
All sound like good ideas to me! 

> (Out of curiosity, how easy is it to notice a new thread in the forum has been posted?)

I get an email with every new Github issue, Discussions post etc., but I might be unusual like that! Let me know if you want me to promote anything on the Discord, and I can send people to threads like this one.

**Comment by HippasusTwo** [comment]
Alright, awhile back I created some animations that would be good for such a minigame. Maybe triggering via an item a little animation to begin with? And then later on making it so on the anim screen you can move it around via keyboard or something and aim for the head of the critter, I don't know how fusion is currently going.
c82863f53b6512263f5f5bf5ccfb4172a30711a7
dc85718ff6a57d0569661300d7b9c149746c692c
fa799a2de4472206bf55f5fbc4a1c4ccd3a64b2b
6daacfe068fbed9acdd870235807d1ce927357aa

Another thing that might be interesting is a follow-me script for starters. You basically just spawn it behind you when there's a free tile and have it mirror your movements, unless you turn around toward it, then to prevent it obstructing you a secondary press must make it swap positions with you.

**Reply by Sanglorian** [reply]
I like these! The only thing I'd suggest is to add some shadows and highlights to the brush and ball-on-stick - to make them feel more three-dimensional.

**Reply by HippasusTwo** [reply]
I may do that at some point.

I did an experimental follow-me script and made some more tuxemon follow sprites for all of the starters except rockitten which already has some.
The main problem with the followme is my attempt to turn off collisions didn't work for some reason, and it causes a crash when other characters try to walk to your character and can't find a path because your critter is there. A 'teleport to near you' function I added is a bit buggy because again, the failure to disable the collision, so the old collision gets left behind and you get an invisible wall. Otherwise it feels pretty good: I make talking to it make it temporarily go away from you so you don't get stuck. The starter sprites don't update with evolution on the script or take into account if the starter is still in the party in the followme.
fc89532066ca6f4c9871a02c2465d57bf0985875

**Reply by Sanglorian** [reply]
Interesting. I'd recommend making the followme monster not provide a collision at all. If that means NPCs stand on top of your monster sometimes, I think that's okay.

**Reply by HippasusTwo** [reply]
I actually tried to disable and it didn't work, so I'd need to dig around (or someone else could do it) to figure out why my attempt at setting collisions off didn't work. It looks like there is a collisions var you are supposed to turn off.
